### PR TITLE
TGUI input for paper strikes again

### PR DESCRIPTION
**Pull request by m-dzianishchyts**
## What Does This PR Do
Enables TGUI input for paper. Works properly now with customizable trim.
We reverted this in #24270, it is better now, I think.

Add one more argument for `/proc/tgui_input_text()` and `/datum/tgui_input_text` constructor, I didn't find any consequences of this. 

## Why It's Good For The Game
We all praise TGUI.

## Testing
Tested for all cases, fine trim, tgui input detected.

image

## Changelog
:cl: Maxiemar
tweak: TGUI input is enabled for paper again, works properly now.
/:cl:

### CODE REVIEW

**Comment by CinnamonSnowball** [comment]
This doesn't seem to actually address the reasons paper's TGUI input was reverted
> TGUI Input Text in the paper was partly a mistake, as many people are used to indenting on Enter rather than Shift - Enter
Plus it had a 1024 character limit, which is pretty silly for paper.

**Comment by Scribble-Sheep** [comment]
Are the two issues mentioned in 24270 still present, out of curiosity?

File: code/__HELPERS/text.dm [hunk 1]
```diff
@@ -54,42 +54,40 @@
...
 
 //Returns null if there is any bad text in the string
-/proc/reject_bad_text(text, max_length=512)
+/proc/reject_bad_text(text, max_length = 512)
 	if(length_char(text) > max_length)	return			//message too long
 	var/non_whitespace = 0
-	for(var/i=1, i<=length_char(text), i++)
-		switch(text2ascii_char(text,i))
-			if(62,60,92,47)	return			//rejects the text if it contains these bad characters: <, >, \ or /
+	for(var/i = 1, i <= length_char(text), i++)
```
Comment (Burzah) [inline code/__HELPERS/text.dm:76]:
```suggestion
	for(var/i in 1 to length_char(text))
```

**Review by Burzah — CHANGES_REQUESTED** [review]

**Comment by FunnyMan3595** [comment]
> TGUI Input Text in the paper was partly a mistake, as many people are used to indenting on Enter rather than Shift - Enter
> Plus it had a 1024 character limit, which is pretty silly for paper.

I'm with the other commenters on this one, both of these need to be addressed before I'm comfortable with this change.

**Pull request closed without merging**